### ISSUE TITLE
Wrong data in lfunc_instances regarding how an L-function factors

### ISSUE THREAD

**Issue by edgarcosta** [LMFDB/lmfdb#2704]
The G2C http://beta.lmfdb.org/Genus2Curve/Q/194084/b/776336/1
factors as
* Elliptic curve 11.a3
* Elliptic curve 17644.a1

however, when one looks in lfunc_instances  (previously we would look into the collection genus2_curves['Lfunctions.instances']) to see how the L-function factors we get:
* EllipticCurve/Q/11/a/
* EllipticCurve/Q/17644/b

See:
http://www.lmfdb.org/api/genus2_curves/Lfunctions.instances/?Lhash=s1167315626697070618
or 
http://beta.lmfdb.org/api/lfunc_instances/?Lhash=1167315626697070618

As a consequence, we display `EC isogeny class 17644.b` as related object in: http://beta.lmfdb.org/Genus2Curve/Q/194084/b/

ps: in a upcoming pull request I tried to use this known factorization and display on the L-function page, and as consequence we also there display the wrong factors: http://cmfs.lmfdb.xyz/L/Genus2Curve/Q/194084/b/

**Comment by JohnCremona** [comment]
Yet again this is caused by LMFDB labels and Cremona labels being different (sometimes).   Isogeny class 17644.b (LMFDB label) is also class 17644a (Cremona label).  The code which looks these up must use the correct label.  It probably can be fixed with a small code change (using the correct key) rather than a data change.

**Comment by AndrewVSutherland** [comment]
@JohnCremona The URL http://beta.lmfdb.org/EllipticCurve/Q/17644/b/ that appears in the lfunc_instances record:

http://beta.lmfdb.org/api/lfunc_instances/9955369

is unambiguous (and wrong).  Either of the URLs http://beta.lmfdb.org/EllipticCurve/Q/17644b (Cremona label) or http://beta.lmfdb.org/EllipticCurve/Q/17644/a/ (LMFDB label) would work, and one of these should be used.

It is easy to check that the data is wrong: if you add the trace hashes for 11.a and 17644.b=17644a (mod 2^61-1) you do not get the Lhash 1167315626697070618 listed in the record, but if you add the trace hashes for 11.a and 17644.a=17644b you do.

That is not to say there wasn't a bug in the original script that created the instance records (which I don't think is even in github) that confused Cremona and LMFDB labels, there probably was, my point is simply that a data change *is* required, but none of the current genus2 or lfunction code needs to change.

Rather than fixing this one record I will write a script to run through all 2681 G2C/ECQP Lfunction instance records and fix this and any similar problems.

**Comment by JohnCremona** [comment]
I wrote the script https://github.com/LMFDB/lmfdb/blob/master/scripts/elliptic_curves/import_ec_lfunction_data.py  so it is in fact my fault, I think the bug is in line 196 of that file.  Apologies for that, and for jumping to the wrong conclusion earlier!

If that is the source of this then a more drastic fix will be needed, i.e. not just for genus 2 - related instances.  If that is needed I have adapted that script for postgres but not tested that yet...

**Comment by AndrewVSutherland** [comment]
@JohnCremona the script you referred looks fine, it is just uploading lfunc instance data for individual elliptic curves over Q, and the data looks correct, in particular, I just checked and the recrod

http://beta.lmfdb.org/api/lfunc_instances/?Lhash=2044405669869775350
http://beta.lmfdb.org/api/lfunc_instances/?Lhash=427549817501221067

are both correct (if you had mistakenly used URLs with Cremona labels formatted as LMFDB labels they would not be).  This problem is specific to lfunc_instance records of type ECQP that correspond to degree 4 L-functions arising from products of elliptic curves that are isogenous to the Jacobian of a genus 2 curve -- these were previously stored in a separate file in the genus2_curves database in mongo and then moved to lfunc_instances in the migration to postgres (the data problem goes back to the original mongo db data and was not introduced by the migration).

In short, I don't think there is anything for you to worry about!

**Comment by JohnCremona** [comment]
OK, I had missed the distinction between ECQ and ECQP types.  Are these types documented anywhere?  The others are easy to guess:

`sage: db.lfunc_instances.distinct('type')
INFO:lfunc_instances@2019-01-28 08:30:02,894: SELECT DISTINCT "type" FROM "lfunc_instances" ORDER BY "type" ran in 11.9749119282s [db_backend.py:298]
[u'BMF',
 u'CMF',
 u'DIR',
 u'ECNF',
 u'ECQ',
 u'ECQP',
 u'G2Q',
 u'HMF',
 u'MaassGL2',
 u'MaassGL3',
 u'MaassGL4',
 u'MaassGSp4']
`
Does ECQP mean "power of an elliptic curve over Q'?

**Comment by AndrewVSutherland** [comment]
I am recomputing all the ECQP lfunc_instances records now (more than 1800 of them were wrong).  This requires updating them one by one, which is very slow (perhaps we need to add a compound index on Lhash/type to lfunc_instances), but should finish in 2-3 hours.  This should have no impact on beta.lmfdb.org other than corrections gradually appearing, e.g.

    http://beta.lmfdb.org/Genus2Curve/Q/2166/a/

is now correct (it previously listed both isogeny classes 57b and 57c because of a Cremona/LMFDB label issue that was definitely **not** the fault of @JohnCremona).

The data error is also present in the mongo db database, but the code apparently hides the problem, which is undoubtedly why it went unnoticed until now.  I won't bother fixing the mongo db data.

**Comment by AndrewVSutherland** [comment]
@JohnCremona ECQP means **product** of elliptic curves over Q (not necessarily a power of a single elliptic curve).

**Comment by JohnCremona** [comment]
Aha!  Sounds like a reason why there should be a list of these types somewhere.

**Comment by AndrewVSutherland** [comment]
Fixed!
